src/plugins/dashboard/public/dashboard_listing/dashboard_listing.tsx

```diff
@@ -0,0 +1,264 @@
+  );
+
+  const deleteItems = useCallback(
+    async (dashboardsToDelete: Array<{ id: string }>) => {
+      try {
+        const deleteStartTime = window.performance.now();
+
+        await Promise.all(
+          dashboardsToDelete.map(({ id }) => {
+            dashboardSessionStorage.clearState(id);
+            return savedObjectsClient.delete(DASHBOARD_SAVED_OBJECT_TYPE, id);
+          })
+        );
+
+        const deleteDuration = window.performance.now() - deleteStartTime;
+        reportPerformanceMetricEvent(pluginServices.getServices().analytics, {
+          eventName: SAVED_OBJECT_DELETE_TIME,
+          duration: deleteDuration,
+          meta: {
+            saved_object_type: DASHBOARD_SAVED_OBJECT_TYPE,
+            total: dashboardsToDelete.length,
+          },
+        });
+      } catch (error) {
+        toasts.addError(error, {
+          title: dashboardListingErrorStrings.getErrorDeletingDashboardToast(),
+        });
+      }
+
+      setUnsavedDashboardIds(dashboardSessionStorage.getDashboardIdsWithUnsavedChanges());
+    },
+    [savedObjectsClient, dashboardSessionStorage, toasts]
+  );
+
+  const editItem = useCallback(
+    ({ id }: { id: string | undefined }) => goToDashboard(id, ViewMode.EDIT),
+    [goToDashboard]
+  );
+  const emptyPrompt = (
+    <DashboardListingEmptyPrompt
+      createItem={createItem}
+      goToDashboard={goToDashboard}
+      unsavedDashboardIds={unsavedDashboardIds}
+      setUnsavedDashboardIds={setUnsavedDashboardIds}
+      useSessionStorageIntegration={useSessionStorageIntegration}
+    />
+  );
+
+  const { getEntityName, getTableListTitle, getEntityNamePlural } = dashboardListingTableStrings;
+
+  return (
+    <I18nProvider>
+      <TableListViewKibanaProvider
+        {...{
+          core: {
+            application: application as TableListViewApplicationService,
+            notifications,
+            overlays,
+            http,
+          },
+          toMountPoint,
+          savedObjectsTagging: savedObjectsTagging.hasApi // TODO: clean up this logic once https://github.com/elastic/kibana/issues/140433 is resolved
+            ? ({
+                ui: savedObjectsTagging,
+              } as TableListViewKibanaDependencies['savedObjectsTagging'])
+            : undefined,
+          FormattedRelative,
+        }}
+      >
+        <TableListView<DashboardSavedObjectUserContent>
+          getDetailViewLink={({ id, attributes: { timeRestore } }) =>
+            getDashboardUrl(id, timeRestore)
+          }
+          deleteItems={!showWriteControls ? undefined : deleteItems}
+          createItem={!showWriteControls ? undefined : createItem}
+          editItem={!showWriteControls ? undefined : editItem}
+          entityNamePlural={getEntityNamePlural()}
+          tableListTitle={getTableListTitle()}
+          headingId="dashboardListingHeading"
+          initialPageSize={initialPageSize}
+          initialFilter={initialFilter}
```

Comment:
I wonder if it also makes sense to expose `TableListViewProps.urlStateEnabled` in the `DashboardListingProps`? Setting `urlStateEnabled: false` appears to be the only way to set the `initialFilter` without including the filter in the url. Without setting to false, I believe the `TableListView` overwrites the `initialFilter` if it does not match (or even exist) in the url.

If we do decide to expose `urlStateEnabled` it might be nice to provide another example in Portable Dashboard Examples showing how to set an initial filter, e.g. `<DashboardListingTable initialFilter="tag:(Security)" urlStateEnabled={false} ... />`. I had to dig in rather deep into code to figure out why I couldn't simply set only the `initialFilter` on the component.